src/pipelinedefinition.cpp

```diff
@@ -546,8 +546,8 @@ class NodeValidator {
         // Affect shape by demultiplexer/gather if applies.
         const auto& tensorInput = this->inputsInfo.at(modelInputName);
         const auto& tensorOutput = this->dependencyOutputsInfo.at(modelOutputName);
-        shape_t tensorInputShape = tensorInput->getShape();
-        shape_t tensorOutputShape = tensorOutput->getShape();
+        shape_t tensorInputShape = tensorInput->getShape_3().getFlatShape();
```

Comment:
Is it only the single model path implementation for now? If so, we should have separate tasks for Validation/Deserialization/Serialization for DAG

Reply:
There is separate task for DAGs